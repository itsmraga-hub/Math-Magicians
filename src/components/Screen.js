// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

export default class Screen extends PureComponent {
  render() {
    const { value } = this.props;
    return (
      <span className="grid-span-full grid-item screen">{value}</span>
    );
  }
}

Screen.propTypes = {
  value: PropTypes.string.isRequired,
};
