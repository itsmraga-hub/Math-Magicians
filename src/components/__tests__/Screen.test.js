import React from 'react';
import renderer from 'react-test-renderer';

import Screen from '../Screen';

it('<Header /> Renders properly', () => {
  const value = '1234';
  const tree = renderer.create(<Screen value={value} />).toJSON();
  expect(tree).toMatchSnapshot();
});
