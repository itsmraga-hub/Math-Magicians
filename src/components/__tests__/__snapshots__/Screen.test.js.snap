// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Renders properly 1`] = `
<span
  className="grid-span-full grid-item screen"
>
  1234
</span>
`;
